macros/src/main/scala/gestalt/macros/Quasiquotes.scala

```diff
@@ -9,8 +9,10 @@ class modsTest extends StaticAnnotation {
     println(mods.toString)
     assert(mods.is(flags.Private))
 
-    // val q"$mods class $name" = q"case class A(x: Int)"
-    // assert(mods.is(flags.Case))
+    val q"$mods1 class $name2 $mods2 ($params)" = q"case class A private[core](x: Int)"
+    assert(mods1.is(flags.Case))
+    assert(mods2.is(flags.Private))
+    assert(mods2.privateWithin == "core")
```

Comment:
This might be the case why it's not recommended to inspect tree structures use quasiquotes -- it's error-prone, it's easy to forget `$mods2` in the above.

Using extractors, macro writers just take primary constructor as a whole, it's simple and reliable.